packages/thrift-client/src/main/connections/HttpConnection.ts

```diff
@@ -217,56 +214,47 @@ export class HttpConnection extends Core.ThriftConnection<RequestOptions> {
                 : this.url
 
         // Merge user options with required options
-        const requestOptions: RequestOptions & UrlOptions = Core.overlayObjects(
-            this.requestOptions,
-            options,
+        const requestOptions: RequestOptions = mergeOptions(
+            mergeOptions(this.requestOptions, options),
             {
                 method: 'POST',
                 body: dataToWrite,
-                encoding: null, // Needs to be explicitly set to null to get Buffer in response body
-                url: requestUrl,
                 headers: {
-                    'Content-Length': dataToWrite.length,
                     'Content-Type': 'application/octet-stream',
                 },
             },
         )
 
         return new Promise((resolve, reject) => {
-            this.requestImpl(
-                requestOptions,
-                (err: any, response: RequestResponse, body: Buffer) => {
+            this.requestImpl(requestUrl, requestOptions)
+                .then(response => {
                     if (
                         shouldRetry(response, retry, this.withEndpointPerMethod)
                     ) {
                         resolve(
                             this.write(dataToWrite, methodName, options, true),
                         )
+                    } else if (isErrorResponse(response)) {
+                        reject(response)
                     } else {
-                        if (hasError(err)) {
-                            reject(err)
-                        } else if (isErrorResponse(response)) {
-                            reject(response)
-                        } else {
+                        response.arrayBuffer().then(arrayBuffer =>
                             resolve({
-                                statusCode: response.statusCode,
+                                statusCode: response.status,
                                 headers: response.headers,
-                                body,
-                            })
-                        }
+                                body: Buffer.from(arrayBuffer),
```

Comment:
> The response.arrayBuffer() function is used to get an ArrayBuffer and convert the binary response into a Node Buffer, which is what the generated code expects to use when sending requests through the connection. We'll need to polyfill that with Webpack, but it should be just fine.

Just to clarify... you're saying `Buffer.from()` needs to be polyfilled? And I guess the whole `Buffer` class? There's not a standard `Buffer` for browsers, is there? But the code has been using `Buffer` for a while, right? So does that mean people using this library are already polyfilling `Buffer`? And the difference now is that they need to make sure their polyfil has a `Buffer.from()` function?

Reply:
Yep, the `Buffer` global class doesn't exist in the browser, obviously, so it has to be polyfilled - webpack 4 used to do this by default, but webpack 5 won't anymore so this is more of a note to myself that I needed to add that into the new webpack config as I tested athena_ui.